src/app/shared/dovetail/dovetail-audio.ts

```diff
@@ -121,15 +115,12 @@ export class DovetailAudio extends ExtendableAudio {
   }
 
   get duration() {
-    if (typeof this.arrangement.duration === 'undefined') {
-      this.arrangement.duration = this.arrangement.entries.reduce(sumDuration, 0);
-    }
-    return this.arrangement.duration;
+    return this.arrangement.map(s => s.duration).reduce((a, b) => a + b, 0);
```

Comment:
It's not actually that expensive to recalculate this every time.  I don't think...

Reply:
Depends on how often it's called since you're making 2 anonymous functions in addition to the math, I think when I originally memoized I was still traumatized from my time in AngularJS where this thing would have been invoked many many times per second and the result here would have been a garbage collection nightmare 😥

Reply:
True.  I was mostly thinking that the main calls to this (in the PlayerComponent) are within the audio `ondurationchange` and `ontimeupdate` events.  And if those are getting called, we would have had to recalculate the currentTime anyways.  Granted ... probably running that map/reduce 1 extra time per change detection cycle.  We'll see how hot my computer gets running 10 of these at once.